File: Namrata-Sanger/88.Merge Sorted Array.js
/*
Day 11
88. Merge Sorted Array
Link: https://leetcode.com/problems/merge-sorted-array/
Level : Easy
*/

/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
// time complexity O(n)
var merge = function(nums1, m, nums2, n) {
    m = m-1;
    let p = nums1.length - 1;
    let q = n - 1;
    while(q >= 0){
        if(nums1[m] < nums2[q]  || m == -1){
            nums1[p] = nums2[q];
            q--;
            p--;
        }else if(nums1[m] >= nums2[q]){
            nums1[p] = nums1[m];
            p--;
            m--;
        }
    }
    return nums1;
};

File: Vindeep-Chaudhari/88_Merge_Sorted_Array.cpp
/**
 * Problem Link - https://leetcode.com/problems/longest-continuous-increasing-subsequence/
 */

class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        
        int num1Ind = m-1;
        int num2Ind = n-1;
        int i = m+n-1;    
        
        while(num1Ind >= 0 && num2Ind >= 0){
            if(nums1[num1Ind]>nums2[num2Ind]) {  
                nums1[i--]=nums1[num1Ind--];
            } else {
                nums1[i--]=nums2[num2Ind--];
            }   
        }
        
        while(num2Ind >= 0) {        
            nums1[i--]=nums2[num2Ind--];
        }

    }
};

// Time complexity - O(m+n)
// Space complexity - O(1)